File: quicksort.c
void swap(int* left, int* right){
    if(*left == *right) return; 
    
    int temp = *right; 
    *right = *left; 
    *left = temp; 
}

int partition(int* array, int start, int end){
    int pivot = array[end];
    int i = start;
    for(int j = start; j < end; j++){
        if(pivot <= array[j]){
            swap(&array[j], &array[i]);
            i++;
        }
    }
    swap(&array[i], &array[end]);
    return i; 
}

void quicksort(int* array, int start, int end){
    if(start >= end) return;

    int pivot = array[end]; 
    int i = partition(array, start, end);
    quicksort(array, start, i-1);
    quicksort(array, i+1, end-1);
}

File: quicksort.py

class QuickSort:
    def __init__(self, input):
        self.list = input

    def partition(self, start, end):
        if start >= end:
            return
        pivot = self.list[end]
        i = start
        for j in range(start, end):
            if pivot > self.list[j]:
                (self.list[i], self.list[j]) = (self.list[j], self.list[i])
                i += 1
        (self.list[i], self.list[end]) = (self.list[end], self.list[i])
        return i

    def sort(self, start, end):
        if start >= end:
            return
        i = self.partition(start, end)
        self.sort(start, i-1)
        self.sort(i+1, end)


if __name__ == "__main__":
    input = [1, 48, 2, 3, 49, 8, 57, 2, 9, 65, 14]
    sort = QuickSort(input)
    sort.sort(0, len(input)-1)
    print(input)